packages/wrangler/src/__tests__/config/findWranglerConfig.test.ts

```diff
@@ -0,0 +1,280 @@
+					configPath: path.resolve(`foo/bar/wrangler.${ext}`),
+					userConfigPath: path.resolve(`foo/bar/wrangler.${ext}`),
+				});
+				expect(findWranglerConfig("./foo/bar/qux")).toEqual({
+					configPath: path.resolve(`foo/bar/wrangler.${ext}`),
+					userConfigPath: path.resolve(`foo/bar/wrangler.${ext}`),
+				});
+				expect(std).toEqual(NO_LOGS);
+			}
+		);
+
+		describe.each([
+			["json", "jsonc"],
+			["json", "toml"],
+			["jsonc", "toml"],
+		])("should prefer the wrangler.%s over wrangler.%s", (ext1, ext2) => {
+			it("in the same directory", async () => {
+				await seed({
+					[`wrangler.${ext1}`]: "DUMMY",
+					[`wrangler.${ext2}`]: "DUMMY",
+				});
+				expect(findWranglerConfig(".")).toEqual({
+					configPath: path.resolve(`wrangler.${ext1}`),
+					userConfigPath: path.resolve(`wrangler.${ext1}`),
+				});
+				expect(std).toEqual(NO_LOGS);
+			});
+
+			it("in different directories", async () => {
+				await seed({
+					[`wrangler.${ext1}`]: "DUMMY",
+					[`foo/wrangler.${ext2}`]: "DUMMY",
+				});
+				expect(findWranglerConfig("./foo")).toEqual({
+					configPath: path.resolve(`wrangler.${ext1}`),
+					userConfigPath: path.resolve(`wrangler.${ext1}`),
+				});
+				expect(std).toEqual(NO_LOGS);
+			});
+		});
+
+		it("should return user config path even if a deploy config is found", async () => {
+			await seed({
+				[`wrangler.toml`]: "DUMMY",
+				[".wrangler/deploy/config.json"]: `{"configPath": "../../dist/wrangler.json" }`,
+				[`dist/wrangler.json`]: "DUMMY",
+			});
+			expect(
+				findWranglerConfig(".", { useRedirectIfAvailable: false })
+			).toEqual({
+				configPath: path.resolve(`wrangler.toml`),
+				userConfigPath: path.resolve(`wrangler.toml`),
+			});
+			expect(std).toEqual(NO_LOGS);
+		});
+	});
+
+	describe("(useRedirectIfAvailable: true)", () => {
+		it("should return redirected config path if no user config and a deploy config is found", async () => {
+			await seed({
+				[".wrangler/deploy/config.json"]: `{"configPath": "../../dist/wrangler.json" }`,
+				[`dist/wrangler.json`]: "DUMMY",
+				["foo/holder.txt"]: "DUMMY",
+			});
+			expect(findWranglerConfig(".", { useRedirectIfAvailable: true })).toEqual(
+				{
+					configPath: path.resolve(`dist/wrangler.json`),
+				}
+			);
+			expect(
+				findWranglerConfig("./foo", { useRedirectIfAvailable: true })
+			).toEqual({
+				configPath: path.resolve(`dist/wrangler.json`),
+			});
+			expect(std).toMatchInlineSnapshot(`
+				Object {
+				  "debug": "",
+				  "err": "",
+				  "info": "",
+				  "out": "",
+				  "warn": "[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mUsing redirected Wrangler configuration.[0m
+
+				  Configuration being used: \\"dist/wrangler.json\\"
+				  Original user's configuration: \\"<no user config found>\\"
+				  Deploy configuration file: \\".wrangler/deploy/config.json\\"
+
+
+				[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mUsing redirected Wrangler configuration.[0m
```

Comment:
is it expected to get a duplicated warning?

Reply:
Oh, probably we test a child dir

Reply:
Yes we are calling `findWranglerConfig()` twice so we expect two warnings in the logs.